src/usr/local/buildpack/utils/cache.sh

```diff
@@ -85,3 +107,82 @@ function get_cache_fill_level () {
   check BUILDPACK_CACHE_DIR true
   df --output=pcent "${BUILDPACK_CACHE_DIR}" | tr -dc '0-9'
 }
+
+# Will calculate the checksum for the given string
+function calculate_checksum () {
+  local text=$1
+  echo "${text}" | sha1sum | awk '{print $1}'
+}
+
+# Will cache the given folder under a given key in the cache
+# Any existing entry under the given key will be overriden
+# If the key is not passed, then the path will be used as key
+#
+# Only enabled when BUILDPACK_CACHE_DIR is set, noop otherwise
+#
+# Will return the path to the cached folder as tar file
+function cache_folder () {
+  local path=$1
+  local key=${2:-${1}}
+
+  check path true
+  check key true
+
+  if [ -z "${BUILDPACK_CACHE_DIR}" ]; then
+    # BUILD_CACHE_DIR is not set
+    return
+  fi
+
+  local checksum
+  checksum=$(calculate_checksum "${key}")
+
+  local filename="${checksum}/folder.tar"
+  local cache_path="${BUILDPACK_CACHE_DIR}/${filename}"
+
+  # create folder
+  mkdir -p "${BUILDPACK_CACHE_DIR}/${checksum}"
+
+  # remove file if it exists
+  if [ -e "${cache_path}" ]; then
+    rm "${cache_path}"
+  fi
+
+  # tar folder
+  tar -cf "${cache_path}" -C "${path}" .
```

Comment:
tgz is very slow, we should use xz compression, or better zstd

Reply:
I am not compressing the files here, I basically just need the folder to be one file, we could use zip or whatever.
I opted for tar without compression.

Reply:
@viceice Do we still need to change this and use compression?

Reply:
yes i think so, as the app otherwise is getting out of Disk space more often

Reply:
👍 will do

Reply:
- install `zstd` with apt_install
- compress: `tar -C $path -cf - . | zstd --no-progress -z -T0 --long=30 -o /tmp/$file`
- uncompress: `zstd --no-progress -c -d -T0 --long=30 /tmp/$file | tar -C $path -xpf -`

Using this on my ci to compress node_modules and dotnet packages folder, is pretty fast.
Maybe we can play with parameters to weigth speed over compression.